src/Microsoft.Identity.Client/Internal/OAuth2/OAuth2Client.cs

```diff
@@ -88,18 +88,34 @@ public void AddBodyParameter(string key, string value)
             }
 
             HttpResponse response = null;
-
-            if (method == HttpMethod.Post)
+            Uri endpointUri = CreateFullEndpointUri(endPoint);
+            var httpEvent = new HttpEvent()
+            {
+                HttpPath = EventBase.ScrubTenant(endpointUri),
+                QueryParams = String.Join("&", MsalHelpers.ParseKeyValueList(endpointUri.Query, '&', false, true, requestContext).Keys)
```

Comment:
We could do that, but that would require the setter (or the HttpEvent class itslef) to take one more parameter `requestContext` (which is a pre-existing requirement for that `ParseKeyValueList()` helper). That would feel weird, wouldn't it?

Again, the bottom line is the 2 ways behave the same. What do you think?